source/libNsdl/src/sn_grs.c

```diff
@@ -1022,14 +984,14 @@ void sn_grs_mark_resources_as_registered(struct nsdl_s *handle)
         return;
     }
 
-    const sn_nsdl_resource_info_s *temp_resource;
+    sn_nsdl_dynamic_resource_parameters_s *temp_resource;
 
     temp_resource = sn_grs_get_first_resource(handle->grs);
 
     while (temp_resource) {
-        if (temp_resource->resource_parameters_ptr) {
-            if (temp_resource->resource_parameters_ptr->registered == SN_NDSL_RESOURCE_REGISTERING) {
-                temp_resource->resource_parameters_ptr->registered = SN_NDSL_RESOURCE_REGISTERED;
+        if (temp_resource->static_resource_parameters) {
```

Comment:
What is the reason for this check? (Only do what function name indicates if static resources are used?) 
Is non-static resources handled elsewhere?